src/arguments/image.ts

```diff
@@ -1,31 +1,42 @@
-import { fetchAvatar } from '#utils/util';
+import { cast, fetchAvatar } from '#utils/util';
+import { Argument, ArgumentContext } from '@sapphire/framework';
 import type { Image } from 'canvas';
-import type { Message } from 'discord.js';
-import { Argument, Possible } from 'klasa';
+import { User } from 'discord.js';
 
-export default class extends Argument {
-	public async run(arg: string, possible: Possible, message: Message): Promise<Image> {
-		// If theres nothing provided, search the channel for an image.
-		if (!arg) {
-			// ! TODO: Re-enable after moving to sapphire!!
-			/*
-			// Configurable minimum of messages
-			const minimum = possible.min || 20;
+export class UserArgument extends Argument<Image> {
```

Comment:
These seems like the perfect opportunity for `ExtendedArgument`:
```ts
@ApplyOptions<ExtendedArgumentOptions<'userName'>>({ baseArgument: 'userName' })
export class UserArgument extends ExtendedArgument<'userName', Image> {
    public async handle(user: User, { argument }: ExtendedArgumentContext): AsyncArgumentResult<Image> {
        try {
            return this.ok(await fetchAvatar(user));
        } catch {
            return this.error(argument, { identifier: 'TBD' });
        }
    }
}
```

Reply:
More like, Kyra trashed 90% of this argument despite the fact that there was a huge TODO: Fix when moving to sapphire.

Reply:
There's no signal to tell Lexure "hey, I got what I wanted from an alternative way, please revert your state!"

Instead, we can do `args.attachment('image')`, `args.attachment('audio-queue')`, and their result variants, which doesn't use Lexure, but the data from the discord `Message`.

Reply:
I don't get it.. so we can't have attachments as arg input?

Reply:
Fixed in a4d1ab9!